Python/crossfire/crossfire/occurrences.py

```diff
@@ -0,0 +1,32 @@
+from queue import Empty, Queue
+
+
+class Occurrences:
+    def __init__(self, client):
+        self.client = client
+        self.buffer = Queue()
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self.buffer.empty():
+            self.load_occurrences()
+
+        try:
+            occurrences = self.buffer.get_nowait()
```

Comment:
Do jeito q você projetou, cada iteração do gerador devolverá uma lista de ocorrências. Isso complica o uso depois, na minna opinião. Vc cogitou que o gerador pode retornar uma por vez?

Caso a ideia seja retornar uma lista de ocorrências, qual a função do _buffer_?

```suggestion
            occurrence = self.buffer.get_nowait()
```